File: examples/python-examples/MeshFillHole.dox.py
import meshlib.mrmeshpy as mrmeshpy

# Load mesh
mesh = mrmeshpy.loadMesh("mesh.stl")

# Find single edge for each hole in mesh
hole_edges = mesh.topology.findHoleRepresentiveEdges()

# Setup filling parameters
params = mrmeshpy.FillHoleParams()
params.metric = mrmeshpy.getUniversalMetric(mesh)

# Alternatively, mrmeshpy.fillHoles(mesh, hole_edges, params) fills all holes at once.
for e in hole_edges:
    # Fill hole represented by `e`
    mrmeshpy.fillHole(mesh, e, params)

# Save result
mrmeshpy.saveMesh(mesh, "filledMesh.stl")

File: examples/js-examples/MeshFillHole.dox.js
import createMeshLib from '@meshinspector/meshlib';

const ml = await createMeshLib();

// Load mesh
using mesh = ml.MeshLoad.fromAnySupportedFormat('mesh.stl');

// Find a single representative edge for each hole in the mesh
using topology = mesh.topology;
const holeEdges = topology.findHoleRepresentiveEdges();

// Setup filling parameters
using params = new ml.FillHoleParams();
using metric = ml.getUniversalMetric(mesh);
params.metric = metric;

// Fill all holes at once (ml.fillHole( mesh, edge, params ) fills a single hole)
ml.fillHoles(mesh, holeEdges, params);

// Save result
ml.MeshSave.toAnySupportedFormat(mesh, 'filledMesh.stl');
